cmd/influx/importer/v8.go

```diff
@@ -0,0 +1,222 @@
+	precision          string
+	writeConsistency   string
+	file, version      string
+	compressed         bool
+}
+
+// NewV8Config returns an initialized *V8Config
+func NewV8Config(username, password, precision, writeConsistency, file, version string, u url.URL, compressed bool) *V8Config {
+	return &V8Config{
+		username:         username,
+		password:         password,
+		precision:        precision,
+		writeConsistency: writeConsistency,
+		file:             file,
+		version:          version,
+		url:              u,
+		compressed:       compressed,
+	}
+}
+
+// V8 is the importer used for importing 0.8 data
+type V8 struct {
+	client                                     *client.Client
+	database                                   string
+	retentionPolicy                            string
+	config                                     *V8Config
+	wg                                         sync.WaitGroup
+	line, command                              chan string
+	done                                       chan struct{}
+	batch                                      []string
+	totalInserts, failedInserts, totalCommands int
+}
+
+// NewV8 will return an intialized V8 struct
+func NewV8(config *V8Config) *V8 {
+	return &V8{
+		config:  config,
+		done:    make(chan struct{}),
+		line:    make(chan string),
+		command: make(chan string),
+		batch:   make([]string, 0, batchSize),
+	}
+}
+
+// Import processes the specified file in the V8Config and writes the data to the databases in chukes specified by batchSize
+func (v8 *V8) Import() error {
+	// Create a client and try to connect
+	config := client.NewConfig(v8.config.url, v8.config.username, v8.config.password, v8.config.version, client.DEFAULT_TIMEOUT)
+	cl, err := client.NewClient(config)
+	if err != nil {
+		return fmt.Errorf("could not create client %s", err)
+	}
+	v8.client = cl
+	if _, _, e := v8.client.Ping(); e != nil {
+		return fmt.Errorf("failed to connect to %s\n", v8.client.Addr())
+	}
+
+	// Validate args
+	if v8.config.file == "" {
+		return fmt.Errorf("file argument required")
+	}
+
+	defer func() {
+		v8.wg.Wait()
+		if v8.totalInserts > 0 {
+			log.Printf("Processed %d commands\n", v8.totalCommands)
+			log.Printf("Processed %d inserts\n", v8.totalInserts)
+			log.Printf("Failed %d inserts\n", v8.failedInserts)
+		}
+	}()
+
+	// Open the file
+	f, err := os.Open(v8.config.file)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	var r io.Reader
+
+	// If gzipped, wrap in a gzip reader
+	if v8.config.compressed {
+		gr, err := gzip.NewReader(f)
+		if err != nil {
+			return err
+		}
+		defer gr.Close()
+		// Set the reader to the gzip reader
+		r = gr
+	} else {
+		// Standard text file so our reader can just be the file
+		r = f
+	}
+
+	// start our accumulator
+	go v8.batchAccumulator()
+
+	// start our command executor
+	go v8.queryExecutor()
+
+	// Get our reader
+	scanner := bufio.NewScanner(r)
+
+	// Process the scanner
+	v8.processDDL(scanner)
+	v8.processDML(scanner)
```

Comment:
Does the `DDL` and `DML` need to be processed in order? Right now it sends it to goroutines that are running concurrently and it seems like they could be run out-of-order.